services/blockchain-indexer/tests/functional/indexer.test.js

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env node
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { ServiceBroker } = require('moleculer');
+
+const { getTableInstance } = require('../../src/database/mysql');
+const accountsIndexSchema = require('../../src/indexer/schema/accounts');
+const blocksIndexSchema = require('../../src/indexer/schema/blocks');
+
+const getAccountIndex = () => getTableInstance('accounts', accountsIndexSchema);
+const getBlocksIndex = () => getTableInstance('blocks', blocksIndexSchema);
+
+const broker = new ServiceBroker({
+	transporter: 'redis://localhost:6379/0',
+	logLevel: 'warn',
+	requestTimeout: 15 * 1000,
+	logger: console,
+});
+
+describe('Test indexer methods', () => {
+	beforeAll(() => broker.start());
+	afterAll(() => broker.stop());
+
+	describe('Connect to indexer and run accounts based process', () => {
+		it('Call indexer.indexAllDelegateAccounts', async () => {
+			const accountsDB = await getAccountIndex();
+			expect(broker.call('indexer.indexAllDelegateAccounts', {})).resolves.not.toThrow();
+
+			const delegates = await accountsDB.find();
+			expect(delegates.length).toBeGreaterThanOrEqual(233); // Delegate accounts on test blockchain
```

Comment:
Kindly consider the following change to ensure the test is deterministic.

```suggestion
			const delegateCount = await accountsDB.count({ isDelegate: true });
			expect(delegateCount).toBeGreaterThanOrEqual(233); // Delegate accounts on test blockchain
```